x/evm/abci.go

```diff
@@ -3,74 +3,48 @@ package evm
 import (
 	"fmt"
 
-	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/ethereum/go-ethereum/common"
 	abci "github.com/tendermint/tendermint/abci/types"
 
+	"github.com/axelarnetwork/axelar-core/utils"
 	"github.com/axelarnetwork/axelar-core/utils/events"
 	"github.com/axelarnetwork/axelar-core/x/evm/types"
 	nexus "github.com/axelarnetwork/axelar-core/x/nexus/exported"
 	"github.com/axelarnetwork/utils/funcs"
+	"github.com/axelarnetwork/utils/slices"
 )
 
-func validateChains(ctx sdk.Context, sourceChainName nexus.ChainName, destinationChainName nexus.ChainName, n types.Nexus) (nexus.Chain, nexus.Chain, error) {
-	sourceChain, ok := n.GetChain(ctx, sourceChainName)
-	if !ok {
-		panic(fmt.Errorf("%s is not a registered chain", sourceChainName))
-	}
-
-	destinationChain, ok := n.GetChain(ctx, destinationChainName)
-	if !ok {
-		return nexus.Chain{}, nexus.Chain{}, fmt.Errorf("%s is not a registered chain", destinationChainName)
-	}
-
-	return sourceChain, destinationChain, nil
-}
-
-func handleTokenSent(ctx sdk.Context, event types.Event, bk types.BaseKeeper, n types.Nexus) bool {
+func handleTokenSent(ctx sdk.Context, event types.Event, bk types.BaseKeeper, n types.Nexus) error {
 	e := event.GetEvent().(*types.Event_TokenSent).TokenSent
 	if e == nil {
 		panic(fmt.Errorf("event is nil"))
 	}
 
-	sourceChain, destinationChain, err := validateChains(ctx, event.Chain, e.DestinationChain, n)
-	if err != nil {
-		bk.Logger(ctx).Info(err.Error())
-		return false
-	}
-
-	sourceCk, err := bk.ForChain(ctx, sourceChain.Name)
-	if err != nil {
-		bk.Logger(ctx).Info(err.Error())
-		return false
-	}
+	sourceChain := funcs.MustOk(n.GetChain(ctx, event.Chain))
+	destinationChain := funcs.MustOk(n.GetChain(ctx, e.DestinationChain))
```

Comment:
why panic instead of return error for destination chain? it is an external input